### DISCUSSION TITLE
Post-filtering process

### DISCUSSION THREAD

**Discussion by obenno** [GoekeLab/xpore#98] [Q&A] [answered]
Hi,

In the method section of paper, you mentioned as:

>To discriminate these positions, we first tested the separation of the two distributions by calculating the probability of the overlapping area of the two clusters. We considered those positions with no more than 50% overlapping area as distinguishable. We also removed those positions where one is inside the other, that is, when more than one intersection point has a density value higher than 0.1. We allow these thresholds to be adjustable by users.

May I ask if these filtrations were performed by `xpore postprocessing` and how to adjust the cutoffs? Thank you!

obenno

**Comment by yuukiiwa** [comment, marked as answer]
Hi obenno,

The filtering steps part you quoted from the method section of the paper are performed when you run `xpore diffmod`, the cutoffs are fixed, which you can find them in `xpore/diffmod/io.py`. `xpore postprocessing` selects the sites based on the majority direction `mod_assignment` of a kmer, and only the sites in their kmer’s majority direction `mod_assignment`  are included from the `xpore postprocessing` output.  For example, if most sites in for the kmer GGACT has a “higher” `mod_assignment`, then xpore postprocessing will remove the GGACT sites with “lower” `mod_assignment`.

Best wishes,
Yuk Kei

**Reply by obenno** [reply]
Hi Yuk Kei,

Thank you very much for your detail explanation, I've found the code chunks for cutoff setup in `xpore/diffmod/io.py`:
https://github.com/GoekeLab/xpore/blob/8222dc14952094c080a451ce35e83aa124643b88/xpore/diffmod/io.py#L363-L367

Best regards,
obenno
